app/main.ipynb

```diff
@@ -31,17 +36,184 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
-   "outputs": [],
    "source": [
-    "# write your code for task 0 here"
+    "df = pd.read_csv(\"../data/adult.csv\")\n",
+    "\n",
+    "df = df.replace(\"?\", pd.NA).dropna()\n",
+    "\n",
+    "valid_salary_values = [\"<=50K\", \">50K\"]\n",
+    "df = df[df[\"salary\"].isin(valid_salary_values)]\n",
+    "\n",
+    "df[\"salary K$\"] = pd.to_numeric(df[\"salary K$\"], errors=\"coerce\")\n",
+    "df = df.dropna(subset=[\"salary K$\"])\n",
+    "\n",
+    "df[\"salary K$\"] = df[\"salary K$\"].astype(int)\n",
+    "\n",
+    "df.to_csv(\"cleaned_data.csv\", index=False)\n",
+    "\n",
+    "df.head()\n"
```

Comment:
you should check data for correctness. So if salary = "<=50K", then you should ensure that column "salary K$" has an appropriate value